tests/snapshots.rs

```diff
@@ -0,0 +1,124 @@
+use serde::Deserialize;
+
+#[derive(Hash, PartialEq, Eq, Deserialize)]
+enum Stage {
+    Vertex,
+    Fragment,
+    Compute,
+}
+
+#[derive(Hash, PartialEq, Eq, Deserialize)]
+struct BindSource {
```

Comment:
We need to find a way to share these structures, at least with `convert.rs` but ideally with Naga backends.
I explored the latter and so far failed, basically got blocked on https://github.com/rust-lang/cargo/issues/4663
Anyway, we can follow-up with that.

Reply:
Agreed, I was also wondering if we should have an `Options` struct per backend (`msl-out` already has `msl::Options`) and serialize/deserialize those options directly

Reply:
Yes, so the problem here is - "when would `Serialize`/`Deserialize` be implemented on these `Option` structs?". Obviously, from Naga point of view, these implementations would be behind *some* feature, be it just `serde`, or the more specific "serialize"/"deserialize". However, from the `convert` example point of view, forcing the user to always enable this feature is an unwanted burden, i.e. we want the users to write:
```
cargo run --example convert --features msl-in,spv-out
```
and not
```
cargo run --example convert --features serde-options,msl-in,spv-out
```
This problem would be easily solved by https://github.com/rust-lang/cargo/issues/4663, but it appears to still be a long way to go...

Reply:
Ah ok, got it. If we remove `convert.rs` and only use it from these snapshot tests, could we do something like `#[cfg_attr(test, derive(Serialize, Deserialize))]`?

Reply:
That's a good idea, we should do this for tests regardless of how "examples/convert.rs" does it, but we'll need to preserve RON format compatibility between the "examples/convert.rs" expectations and the tests.